### PR TITLE
This PR will enable all controller and API routes for TRIPS

### LINKED ISSUE #11: Set up Server API end points in routes.js

### PR DESCRIPTION
**Pull request by ryandhaase**
- Completed: Specs for Trip routes and controller methods
- Refactored Controller methods to make use of Mongoose callbacks
- Set up and installed Chai & Mocha for test suite

fixes #11 #14

### CODE REVIEW

File: public/client/index.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
```
Comment (LiaRowan) [inline public/client/index.html]: This file was removed and should not be included in this PR

File: server/trips/tripmodel.js [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
-var Schema = mongoose.Schema;
-​
-var tripSchema = new Schema({
+
+var tripSchema = new mongoose.Schema({
   tripName: String,
   members: {
     type: Array,
-    default:[]
+    default:[],
```
Comment (LiaRowan) [inline server/trips/tripmodel.js:7]: This is the last property in the object, so the comma should not have been added.

File: server/trips/tripmodel.js [hunk 3]
```diff
@@ -1,24 +1,20 @@
...
   location: String,
   dates: {
     start: Date,
     end: Date,
-    },
+  },
   lists:{
     type:Array,
-    default:[]
+    default:[],
```
Comment (LiaRowan) [inline server/trips/tripmodel.js:16]: This is the last property in the object, so the comma should not have been added.

File: specs/tripSpecs/tripSpecs.js [hunk 4]
```diff
@@ -1,2 +0,0 @@
-var express = require('express');
```
Comment (LiaRowan) [inline specs/tripSpecs/tripSpecs.js:1]: Is this an empty file now?

File: public/client/index.html (hunk 1 above)
Comment (ryandhaase) [inline public/client/index.html]: When we rebase from the previous PR, the file is still there

File: server/trips/tripmodel.js (hunk 2 above)
Comment (ryandhaase) [inline server/trips/tripmodel.js:7]: This is best practices in JS and many companies:
https://github.com/airbnb/javascript see section 19.2
This is the style guide I've been following since November and happens to be the same setup as jshint for atom :)

File: server/trips/tripmodel.js (hunk 3 above)
Comment (ryandhaase) [inline server/trips/tripmodel.js:16]: Seew above^

File: specs/tripSpecs/tripSpecs.js (hunk 4 above)
Comment (ryandhaase) [inline specs/tripSpecs/tripSpecs.js:1]: We removed this file.  It wasn't in the commit so my assumption is this will disappear with this PR
